File: samples/make_happy/make_happy.py

class Cat(object):
    def __init__(self, name):
        self.name = name
        self.happy = False

    def make_happy(self):
        print "{} is becoming happy...".format(self.name)
        self.happy = True

    def is_happy(self):
        print "{} is happy: {}".format(self.name, self.happy)
        return self.happy

if __name__ == '__main__':
    cat = Cat('Ruby')
    cat.is_happy()
    cat.make_happy()
    cat.is_happy()

File: samples/ruby/make_happy.rb

class Cat
  def initialize(name)
    @name = name
    @happy = false
    return
  end

  def make_happy
    print "#{@name} is becoming happy...\n"
    @happy = true
    return
  end

  def is_happy()
    print "#{@name} is happy: ", @happy, "\n"
    @happy
  end
end

cat = Cat.new 'Ruby'
cat.is_happy
cat.make_happy
cat.is_happy

